tests/unit_test/app_common/widgets/streaming_test.py

```diff
@@ -96,3 +98,93 @@ def test_invalid_send_analytic_dxo(self, comp, dxo, fl_ctx, expected_error, expe
+    kwargs = kwargs["kwargs"] if kwargs and kwargs["kwargs"] else {}
+    if global_step is not None:
+        if not isinstance(global_step, int):
+            raise TypeError(f"Expect global step to be an instance of int, but got {type(global_step)}")
+        kwargs[TrackConst.GLOBAL_STEP_KEY] = global_step
+    global_step = kwargs.get(TrackConst.GLOBAL_STEP_KEY, None)
+    dxo = create_analytic_dxo(tag=tag, value=value, data_type=data_type, writer=LogWriterName.TORCH_TB, **kwargs)
+    return dxo
+
+
+ANALYTICS_SENDER_TEST_CASES = [
+    (
+        "text",
+        "textsample",
+        AnalyticsDataType.TEXT,
+        None,
+        {},
+        "ANALYTIC",
+        {"track_key": "text", "track_value": "textsample"},
+        {"analytics_data_type": AnalyticsDataType.TEXT, "tracker_key": LogWriterName.TORCH_TB},
+    ),
+    (
+        "text",
+        "textsample",
+        AnalyticsDataType.TEXT,
+        2,
+        None,
+        "ANALYTIC",
+        {"track_key": "text", "track_value": "textsample", "global_step": 2, "analytics_kwargs": {"global_step": 2}},
+        {"analytics_data_type": AnalyticsDataType.TEXT, "tracker_key": LogWriterName.TORCH_TB},
+    ),
+    (
+        "text",
+        "textsample",
+        AnalyticsDataType.TEXT,
+        None,
+        {"global_step": 3, "extra_arg": 4},
+        "ANALYTIC",
+        {
+            "track_key": "text",
+            "track_value": "textsample",
+            "global_step": 3,
+            "analytics_kwargs": {"global_step": 3, "extra_arg": 4},
+        },
+        {"analytics_data_type": AnalyticsDataType.TEXT, "tracker_key": LogWriterName.TORCH_TB},
+    ),
+    (
+        "set_tag_key_tag_name",
+        "tagvalue",
+        AnalyticsDataType.TAG,
+        None,
+        None,
+        "ANALYTIC",
+        {"track_key": "set_tag_key_tag_name", "track_value": "tagvalue"},
+        {"analytics_data_type": AnalyticsDataType.TAG, "tracker_key": LogWriterName.TORCH_TB},
+    ),
+    (
+        "log_metric_key_name",
+        2.4,
+        AnalyticsDataType.METRIC,
+        20,
+        None,
+        "ANALYTIC",
+        {
+            "track_key": "log_metric_key_name",
+            "track_value": 2.4,
+            "global_step": 20,
+            "analytics_kwargs": {"global_step": 20},
+        },
+        {"analytics_data_type": AnalyticsDataType.METRIC, "tracker_key": LogWriterName.TORCH_TB},
+    ),
+]
+
+
+class TestAnalyticsSender:
+    @pytest.mark.parametrize(
+        "tag,value,data_type,global_step,kwargs,expected_dxo_data_kind,expected_dxo_data,expected_dxo_meta",
+        ANALYTICS_SENDER_TEST_CASES,
+    )
+    def test_add(
+        self, tag, value, data_type, global_step, kwargs, expected_dxo_data_kind, expected_dxo_data, expected_dxo_meta
+    ):
+        print(kwargs)
+        dxo = mock_add(tag=tag, value=value, data_type=data_type, global_step=global_step, kwargs=kwargs)
```

Comment:
Thanks for adding the test.

Can you add test cases like:

```dxo = mock_add(tag=tag, value=value, data_type=data_type, global_step=global_step, **kwargs)```

Because kwargs should be any random dictionary

For example:
```
def cool_f(x, *args, **kwargs):
    if args:
        print(f"args is {args}")

    if kwargs:
        print(f"kwargs is {kwargs}")

input_dict = {"global_step": 100}

cool_f(x=1, a=3, b=10)
cool_f(x=2, global_step=100)
cool_f(x=5, **input_dict)

```

Reply:
should be fixed now